### PR TITLE
Prune framework tests (skip tests for other frameworks if changes are local to another framework)

### PR DESCRIPTION
**Pull request by SamMousa**
Ready for review.

It should bring back test pruning, also newer versions of PHP now use pcov for coverage, increasing run speed a lot.

### CODE REVIEW

**Comment by SamMousa** [comment]
@Naktibalda @DavertMik please review when you have time.

File: .travis.yml [hunk 1]
```diff
@@ -26,28 +24,35 @@ env:
...
-    env: FRAMEWORK=Codeception SUITES=cli,unit,coverage TEST_PATH=. XDEBUG=1 PECL=mongodb
+  # Run our coverage suite on 7.2 (so it uses pcov)
+  - php: 7.2
+    env: FRAMEWORK=Codeception SUITES=cli,unit,coverage TEST_PATH=. PECL="mongodb" COVERAGE=1
   exclude:
-  - php: 7.1
-    env: FRAMEWORK=Codeception SUITES=cli,unit TEST_PATH=. XDEBUG=1 PECL=mongodb
+
+  - php: 5.6
+    env: FRAMEWORK=Yii2 TEST_REPO="https://github.com/Codeception/yii2-tests"
```
Comment (Naktibalda) [inline .travis.yml]: What does this line do? I looked at the last build and Yii2 tests were executed on all versions of PHP.

File: PruneTest.php [hunk 2]
```diff
@@ -50,14 +52,19 @@ function stderr($message)
 $frameworks = [];
 foreach ($files as $file) {
     $match = false;
+    stderr("Testing file: $file");
     foreach ($regexes as $framework => $regex) {
+        stderr("Checking framework $framework...", false);
```
Comment (Naktibalda) [inline PruneTest.php]: Please remove `Yii1` line from `$regexes`, that module is gone.

**Comment by SamMousa** [comment]
> Please make some change in framework module to demonstrate that pruning is actually working.

I can't do that until this is merged; since it always runs all tests when the core changes

File: .travis.yml (hunk 1 above)
Comment (SamMousa) [reply .travis.yml]: I'll try to remove it again; I had some issues with that specific combination when running tests.

Comment (SamMousa) [reply .travis.yml]: The advanced app template requires PHPUnit 6.5, which does not support PHP5.6.
So solutions are:
- We no longer test the advanced app template for any Codeception version.
- We no longer test the Yii2 module for PHP 5.6.
- We adapt the module test to dynamically include / exclude the advanced app template depending on PHP versions.

Comment (SamMousa) [reply .travis.yml]: https://github.com/yiisoft/yii2-app-advanced/issues/440

**Comment by DavertMik** [comment]
Seems good to me! I like it

**Pull request merged**